tests/unit/store/v2/test_channel_map.py

```diff
@@ -0,0 +1,319 @@
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from testtools.matchers import Equals, HasLength, Is, IsInstance
+
+from snapcraft.storeapi.v2 import channel_map
+from tests import unit
+
+
+class ProgressiveTest(unit.TestCase):
+    def test_progressive(self):
+        payload = {"key": "my-key", "paused": False, "percentage": 83.3}
+
+        p = channel_map.Progressive.unmarshal(payload)
+
+        self.expectThat(repr(p), Equals(f"<Progressive: 83.3>"))
+        self.expectThat(p.key, Equals(payload["key"]))
+        self.expectThat(p.paused, Equals(payload["paused"]))
+        self.expectThat(p.percentage, Equals(payload["percentage"]))
+        self.expectThat(p.marshal(), Equals(payload))
+
+    def test_none(self):
+        payload = {"key": None, "paused": None, "percentage": None}
+
+        p = channel_map.Progressive.unmarshal(payload)
+
+        self.expectThat(repr(p), Equals(f"<Progressive: None>"))
+        self.expectThat(p.key, Equals(payload["key"]))
+        self.expectThat(p.paused, Equals(payload["paused"]))
+        self.expectThat(p.percentage, Equals(payload["percentage"]))
+        self.expectThat(p.marshal(), Equals(payload))
+
+
+class MappedChannelTest(unit.TestCase):
+    def setUp(self):
+        super().setUp()
+
+        self.payload = {
+            "architecture": "amd64",
+            "channel": "latest/stable",
+            "expiration-date": None,
+            "progressive": {"key": None, "paused": None, "percentage": None},
+            "revision": 2,
+        }
+
+    def test_channel(self):
+        mc = channel_map.MappedChannel.unmarshal(self.payload)
+
+        self.expectThat(
+            repr(mc),
+            Equals(
+                "<MappedChannel: 'latest/stable' for revision 2 and architecture 'amd64'>"
+            ),
+        )
+        self.expectThat(mc.channel, Equals(self.payload["channel"]))
+        self.expectThat(mc.revision, Equals(self.payload["revision"]))
+        self.expectThat(mc.architecture, Equals(self.payload["architecture"]))
+        self.expectThat(mc.progressive, IsInstance(channel_map.Progressive))
+        self.expectThat(mc.expiration_date, Is(None))
+        self.expectThat(mc.marshal(), Equals(self.payload))
+
+    def test_channel_with_expiration(self):
+        date_string = "2020-02-11T17:51:40.891996Z"
+        self.payload.update({"expiration-date": date_string})
+
+        mc = channel_map.MappedChannel.unmarshal(self.payload)
+
+        self.expectThat(
+            repr(mc),
```

Comment:
it's interesting that you are testing reprs, are they being used for anything other than debugging/development?

Reply:
yes, this is for debugging/devel